### PR TITLE
docs: convert markdown issue templates to issue forms

### PR DESCRIPTION
**Pull request by Anmol-Baranwal**
closes #1234 

## Changes proposed

I've added five issue forms in total and removed existing markdown templates.

I've used the existing templates as inspiration (especially the acceptance criteria).

- Documentation
- Feature Request
- Refactor Code/Feature Improvement
- Bug Report
- Other (to avoid creating blank issues) - discussion

You can see how these will [preview](https://github.com/Anmol-Baranwal/taipy/tree/issue_forms/.github/ISSUE_TEMPLATE) on my fork branch. You cannot see it while creating an issue (which I've enabled) because it's in another branch and not in the `main` branch.

If you want to see how the final issue forms will look in general, then you can check [this](https://github.com/dailydotdev/daily/issues/new?assignees=&labels=Type%3A+Feature&projects=&template=---feature-request.yaml&title=%F0%9F%A7%A9+FEATURE+REQUEST%3A++) or [this](https://github.com/rupali-codes/LinksHub/issues/new?assignees=&labels=enhancement%2Cgoal%3A+addition%2Cpriority%3A+high&projects=&template=feature_request.yml&title=%5BFeature%5D+%3Cwrite+a+small+description+here%3E). I've contributed it myself a few months back.

## Why Issue forms

- This will give you a consistent info so it will be easier to check the issue.
- Even a user with no markdown knowledge can easily fill up the form.

## Note to reviewers

1. Sections that are required are marked with `*` Asterisk.

> For instance.
required

2. I've also added a bit of Markdown content to improve context.

3. I've used labels of existing templates except for `discussion` and `documentation`. Let me know if any changes are necessary in that.

---

@jrobinAV 
Please suggest any necessary changes, and I will make them!

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (florian-vuillemot) [inline .github/ISSUE_TEMPLATE/bug-report.yml:97]:
```suggestion
      placeholder: ex. 3.1.1
```

**Review by florian-vuillemot — DISMISSED** [review]

**Comment by Anmol-Baranwal** [comment]
@florian-vuillemot 
I've updated the placeholder as per suggestion and marked the conversation resolved.

**Review by jrobinAV — DISMISSED** [review]

**Review by florian-vuillemot — DISMISSED** [review]

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/bug-report.yml]:
```suggestion
        - The more details we have, the easier it would be to fix it. 
```

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/bug-report.yml]:
```suggestion
        - If any heading is not applicable, please set it to `NA`.
```

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/bug-report.yml]:
```suggestion
      placeholder: The description of the issue you experienced in detail, including any relevant information or context.
```

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/bug-report.yml]:
```suggestion
      placeholder: The description of the behavior that you expected when encountering the issue.
```

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/bug-report.yml]:
```suggestion
        2. And/or configuration files or code
```

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/bug-report.yml]:
```suggestion
      placeholder: The description of the expected behavior of the problem.
```

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/bug-report.yml]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/bug-report.yml]:
```suggestion
        - label: "Create a related issue in taipy-doc for documentation and Release Notes if relevant."
```

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/bug-report.yml]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/documentation.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/documentation.yml]:
```suggestion
description: "Use this form to present your suggestions on how to improve our docs."
```

File: .github/ISSUE_TEMPLATE/documentation.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/documentation.yml]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/feature-request.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/feature-request.yml]:
```suggestion
      placeholder: A detailed description of the new feature request.
```

File: .github/ISSUE_TEMPLATE/feature-request.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/feature-request.yml]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/feature-request.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/feature-request.yml]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/feature-request.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/feature-request.yml]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/other.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/other.yml]:
```suggestion
description: Use this for any other question or issue. Please do not create blank issues.
```

File: .github/ISSUE_TEMPLATE/other.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/other.yml]:
```suggestion
      description: By submitting this issue, you agree to follow our [Code of Conduct](https://github.com/Avaiga/taipy/blob/develop/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/Avaiga/taipy/blob/develop/CONTRIBUTING.md).
```

File: .github/ISSUE_TEMPLATE/other.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/other.yml]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/refactor-code.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/refactor-code.yml]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/refactor-code.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/refactor-code.yml]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/refactor-code.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/refactor-code.yml]:
```suggestion
      description: By submitting this issue, you agree to follow our [Code of Conduct](https://github.com/Avaiga/taipy/blob/develop/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/Avaiga/taipy/blob/develop/CONTRIBUTING.md).
```

File: .github/ISSUE_TEMPLATE/refactor-code.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/refactor-code.yml]:
```suggestion
        - label: "I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+)."
```

File: .github/ISSUE_TEMPLATE/refactor-code.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/refactor-code.yml]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/bug-report.yml]:
```suggestion
        - label: "I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+)."
```

File: .github/ISSUE_TEMPLATE/documentation.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/documentation.yml]:
```suggestion
        - label: "I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+)."
```

File: .github/ISSUE_TEMPLATE/feature-request.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/feature-request.yml]:
```suggestion
        - label: "I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+)."
```

File: .github/ISSUE_TEMPLATE/other.yml
Comment (FabienLelaquais) [inline .github/ISSUE_TEMPLATE/other.yml]:
```suggestion
        - label: "I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+)."
```

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Mainly minor typographical change requests for better homogeneity.
Great work!

**Comment by Anmol-Baranwal** [comment]
@FabienLelaquais
I've made the changes and marked the suggestions as resolved. 
Let me know if it needs any other changes!

**Review by jrobinAV — DISMISSED** [review]
Thank you @Anmol-Baranwal for your help.

**Comment by Anmol-Baranwal** [comment]
Hi Team, if everything is okay, please review and merge this PR.

**Comment by jrobinAV** [comment]
The two other reviewers are currently on vacation. They should be back in a couple of days. I am waiting for a second approval, and once it's ok, I'll merge it. 

Thank you again.

**Review by florian-vuillemot — DISMISSED** [review]

**Comment by Anmol-Baranwal** [comment]
Hi Team!
Sorry for disturbing you during your vacation time. Please feel free to reply whenever you're available.

One of the tests failed due to a spelling error in `documentation.` Should I correct it, even though it will dismiss the reviews?

image

Let me know your decision. I'm unsure about the issue with the other one.

Have a great day!

**Comment by jrobinAV** [comment]
Hello, 

Oh, right. If you can fix it, that would be great. Thank you.

**Comment by Anmol-Baranwal** [comment]
@jrobinAV 
I kind of missed it.

Anyway, I have updated it with the documentation label present in the repo.

image

I hope this works!

> Let me know if you want me to fix anything else or if any test is failing.

**Review by jrobinAV — APPROVED** [review]

**Review by florian-vuillemot — APPROVED** [review]
Good job thanks!

**Review by FabienLelaquais — APPROVED** [review]
Looks really good! Thanks!!

**Pull request merged**
